src/backend/catalog/pg_attribute_encoding.c

```diff
@@ -972,7 +972,7 @@ check_attribute_encoding_entry_exist(Oid relid, bool *attnum_entry_present)
 
 	Assert(OidIsValid(relid));
 
-	MemSet(attnum_entry_present, false, sizeof(attnum_entry_present));
+	MemSet(attnum_entry_present, false, attnum_entry_present_size);
```

Comment:
MaxHeapAttributeNumber is define, then why not simply
```suggestion
	MemSet(attnum_entry_present, false, MaxHeapAttributeNumber);
```

Reply:
It seemed wrong to just assume that this function would only ever be used on arrays of this size (and not less, for example). I can change this to use a constant, but I think it's cleaner to specify the size explicitly, since this is a static function anyway and this doesn't cost us anything.

Reply:
advanced suggestion
```suggestion
	MemSet(attnum_entry_present, false, sizeof(*attnum_entry_present) * MaxHeapAttributeNumber);
```

Reply:
On second thought, it seems unlikely this function will be used for anything else. I will change it to this and add a comment.